Jenkinsfile

```diff
@@ -13,277 +13,35 @@
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:latest")
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:${version}-slim")
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:latest-slim")
-            }
-
-            response = OsTools.runSafe(this, """
-            curl -X POST https://hub.docker.com/v2/users/login/ \
-              -H 'cache-control: no-cache' -H 'content-type: application/json' \
-              -d '{ "username": "${env.DOCKERHUB_API_USERNAME}", "password": "${env.DOCKERHUB_API_PASSWORD}" }'
-            """)
-            token = readJSON text: response
-            dockerHubApiToken = token.token
-
-            def readme = readFile file: 'README.md', encoding: 'UTF-8'
-            readme = readme.replaceAll("(?s)<!--.*?-->", "")
-            readme = readme.replace("\"", "\\\"")
-            readme = readme.replace("\n", "\\n")
-            readme = readme.replace("\\\$", "\\\\\$")
-            response = httpRequest customHeaders: [[name: 'authorization', value: "JWT ${dockerHubApiToken}"]],
-                acceptType: 'APPLICATION_JSON', contentType: 'APPLICATION_JSON', httpMode: 'PATCH',
-                requestBody: "{ \"full_description\": \"${readme}\" }",
-                url: "https://hub.docker.com/v2/repositories/${organization}/${dockerHubRepository}/"
-          }
-        }
-      }
-      stage('Push tags') {
-        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
-                          usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
-          OsTools.runSafe(this, "git tag ${version}")
-          OsTools.runSafe(this, """
-            git push \
-            https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github.com/${organization}/${gitHubRepository}.git \
-              ${version}
-          """)
-        }
-        OsTools.runSafe(this, "git tag -d ${version}")
-      }
-    }
-  } finally {
-    OsTools.runSafe(this, "docker logout")
-    OsTools.runSafe(this, "docker system prune -a -f")
-    OsTools.runSafe(this, 'git clean -f && git reset --hard origin/master')
-  }
-}
-
-def readVersion() {
-  def content = readFile 'Dockerfile'
-  for (line in content.split('\n')) {
-    if (line.startsWith('ARG IQ_SERVER_VERSION=')) {
-      return getShortVersion(line.substring(22))
-    }
+@Library(['private-pipeline-library', 'jenkins-shared@test-expectations', 'iq-pipeline-library']) _
+
+import com.sonatype.jenkins.shared.Expectation
+
+dockerizedBuildPipeline(
+  lint: {
+    hadolint(['Dockerfile'])
+  },
+  buildAndTest: {
+    validateExpectations([
+          new Expectation('javaVersion', 'java', '-version', 'openjdk version "1.8.0_362"'),
+          new Expectation('userGroups', 'id', '', 'uid=1000(nexus) gid=1000(nexus) groups=1000(nexus)'),
```

Comment:
I think you want `id nexus` otherwise this is likely to return the info on the jenkins user that is executing the command.

Reply:
> I think you want `id nexus` otherwise this is likely to return the info on the jenkins user that is executing the command.

Thank you! That fixed it. 143d9e5